lib/Model/DiagnosticsParser.php

```diff
@@ -17,21 +17,30 @@ public function parse(string $jsonString): array
     {
         $decoded = $this->decodeJson($jsonString);
         $diagnostics = [];
-
-        foreach ($decoded['files'] ?? [] as $fileDiagnostics) {
+         
+        foreach ($decoded['files'] ?? [] as $uri => $fileDiagnostics) {
+            $lines = file($uri);
+            if ($lines === false) {
+                $lines = [];
+            }
+                 
             foreach ($fileDiagnostics['messages'] as $message) {
                 $lineNo = (int)$message['line'] - 1;
                 $lineNo = (int)$lineNo > 0 ? $lineNo : 0;
-
+                $line = $lines[$lineNo] ?? "";
+                 
+                $matches = [];
+                $offset = (\preg_match("/^\\s+/", $line, $matches) === 1) ? mb_strlen($matches[0]) : 0;
```

Comment:
I created a utility for this: https://github.com/phpactor/text-document/blob/master/lib/Util/LineColRangeForLine.php might be better to use that

Reply:
`(new LineColRangeForLine())->rangeFromLine($text, $lineNo)`

Reply:
For some reason I don't have this in my version of text-document.

Reply:
yes you probably need to bump to the next patch version

Reply:
`"phpactor/text-document": "^1.2.4"`

Reply:
`"phpactor/language-server-extension": "^0.6.1",` this is bringing text-document, but it doesn't get me the latest.

Reply:
(I suck at this)

Reply:
yes you need to explicitly require it in composer.json :)

Reply:
Yes, that worked. Now the util didn't do what I needed, debugging now.

Reply:
Why does this add 1 to start?
```
       return new LineColRange(
            new LineCol($lineNo, $start + 1),
            new LineCol($lineNo, $end)
        );
```

Reply:
good question

Reply:
not critical here i guess and can be fixed independently if it is wrong

Reply:
As far as I understand it, it assumes that the line is passed as 1-based index, so it returns start and end as 1 based indices as well. That means probably that end should be `-1` as well. I leave that to you. In this code if I skip to substract 1 from end, the worst case scanerio is that the end will be 1 more than the end of line, which VS will truncate.

Reply:
Just to make things interesting the LSP Position is 0 based.

Reply:
That is why I removed 1 :-)